### PR TITLE
High-order BCs for Upwind 3rd-Order composite operator

### PR DESCRIPTION
**Pull request by valeriabarra**
This is my attempt at adding BCs for the `Upwind3rdOrderBiasedProductC2F` operator that so far was supported only on periodic meshes.

### CODE REVIEW

File: src/Operators/finitedifference.jl [hunk 1]
```diff
@@ -873,7 +878,20 @@ U(v,x)[i] = \\begin{cases}
 This stencil is based on [WickerSkamarock2002](@cite), eq. 4(a).
 
 Supported boundary conditions are:
-- At the moment this is defined only for periodic meshes.
+- [`SetValue(x₀)`](@ref): set the value of `x` to be `x₀` on the boundary. On
+  the left boundary the stencil is
+  ```math
+  U(v,x)[\\tfrac{1}{2}] = \\begin{cases}
+   v[\\tfrac{1}{2}] \\left(8 x_0 + 4 x[1+\\tfrac{1}{2}] \\right) / 12  \\textrm{, if } v[i] > 0 \\\\
+   v[\\tfrac{1}{2}] \\left(4 x_0 + 10 x[1+\\tfrac{1}{2}] -2 x[1+\\tfrac{3}{2}]  \\right) / 12  \\textrm{, if } v[i] < 0
```
Comment (valeriabarra) [inline src/Operators/finitedifference.jl:886]: Not sure if the docs here are a good representation of the code or if this is more aspirational

File: test/Operators/finitedifference/column.jl [hunk 2]
```diff
@@ -675,14 +675,118 @@ end
...
+    end
+
+    # Check convergence rate
+    conv_adv_wc = convergence_rate(err_adv_wc, Δh)
+
+    # Upwind3rdOrderBiasedProductC2F conv, with f(z) = sin(z)
+    @test err_adv_wc[3] ≤ err_adv_wc[2] ≤ err_adv_wc[1] ≤ 5e-2
+    @test conv_adv_wc[1] ≈ 0.5 atol = 0.1
+    @test conv_adv_wc[2] ≈ 0.5 atol = 0.1
+    @test conv_adv_wc[3] ≈ 0.5 atol = 0.1
```
Comment (valeriabarra) [inline test/Operators/finitedifference/column.jl:735]: @simonbyrne , unfortunately, both with `SetValue` and `Extrapolate` BCs, we now have only 0.5 convergence rate. So I am afraid we have to revisit the way we handled the boundary cases, as this is supposed to be higher-order  :/

**Comment by simonbyrne** [comment]
For the boundaries, instead of defining it for `SetValue`, we want to have options that allow setting the 1st interior face (level 1+half) as either:
- 1st order upwind (e.g. what you would get by `UpwindBiasedProductC2F`
- 3rd order one-sided (e.g. what you would get by using the right-biased reconstruction on the right boundary, and vice-versa)

**Comment by valeriabarra** [comment]
@simonbyrne I tried to apply your suggestions to the best of my understanding but obviously I am missing something. 
I tried to use `stencil_interior(RightBiasedC2F()....)` in the `stencil_left_boundary` function both at the boundary point and the first interior one. If we use it in the first interior one, I am not sure I understand how to set the other ones then

**Comment by valeriabarra** [comment]
@simonbyrne do you know what is left to do here, according to what @OsKnoth suggested?
It would be good if this could be merged by tomorrow's meeting.

**Review by simonbyrne — APPROVED** [review]

**Pull request merged**